src/zjs_spi.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017, Intel Corporation.
+
+#ifndef __zjs_spi_h__
+#define __zjs_spi_h__
+
+#include "jerryscript.h"
+
+enum spi_topology {
+    ZJS_TOPOLOGY_FULL_DUPLEX,
+    ZJS_TOPOLOGY_READ,
+    ZJS_TOPOLOGY_WRITE,
+    ZJS_TOPOLOGY_MULTIPLEXED,
+    ZJS_TOPOLOGY_DAISY_CHAIN
+};
+
+enum spi_direction {
+    ZJS_SPI_DIR_READ,
```

Comment:
It seems topology has the same mapping as direction?  READ == READ, WRITE == WRITE, and all others = READ/WRITE?    Maybe you can just consolidate into one enum?